cpp/util/read_key.cc

```diff
@@ -31,8 +34,9 @@ static void EngineFree() {
 static void EngineFinish() {
   ENGINE_finish(engine);
 }
+#endif
 
-util::StatusOr<EVP_PKEY*> ReadPrivateKey(const std::string& file) {
+util::StatusOr<EVP_PKEY*> ReadPrivateKey(const std::string& file, const std::string& engine_name) {
```

Comment:
This is kinda weird since we'll have to change all the call sites to pass in strings whether they want an engine or not.

Would it make more sense to break the engine initialisation stuff out to another function which gets called exactly once at program start up time?